google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/GoogleUserModelItem.java

```diff
@@ -225,6 +230,35 @@ public int getChildCount() {
     return Math.toIntExact(getFilteredChildren().count());
   }
 
+  @Override
+  public void remove(int childIndex) {
+    if (children == null) {
+      // Preserved functionality from default getChildAt() implementation.
+      throw new ArrayIndexOutOfBoundsException("node has no children");
+    }
+
+    // This is different from the default implementation because it uses the raw children vector
+    // instead of relying on the getChildAt() method, which has been modified to only return
+    // filtered children.
+    MutableTreeNode child = (MutableTreeNode) children.elementAt(childIndex);
+    children.removeElementAt(childIndex);
+    child.setParent(null);
+  }
+
+  @Override
+  public void removeAllChildren() {
+    if (children == null) {
+      return;
+    }
+
+    // This is different from the default implementation because it uses the raw children vector
+    // instead of relying on the getChildCount() method, which has been modified to only return
+    // the number of filtered children.
+    for (int i = children.size() - 1; i >= 0; i--) {
```

Comment:
nit: `forEach` instead?

Reply:
It would probably throw a `ConcurrentModificationException` since the loop modifies the underlying `children` vector. Also, I believe the point of reverse-iterating through the children is to prevent any shifting of positions that the `Vector` might do if I removed from beginning to end.